plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -783,7 +783,27 @@ public void createSchema(ConnectorSession session, String schemaName, Map<String
     @Override
     public void dropSchema(ConnectorSession session, String schemaName)
     {
+        Optional<Path> location = metastore.getDatabase(schemaName)
+                .orElseThrow(() -> new SchemaNotFoundException(schemaName))
+                .getLocation()
+                .map(Path::new);
+
         metastore.dropDatabase(new HiveIdentity(session), schemaName);
+
+        location.ifPresent(path -> {
+            try {
+                FileSystem fs = hdfsEnvironment.getFileSystem(new HdfsContext(session), path);
+                // If no files in schema directory, delete it
+                if (!fs.listFiles(path, false).hasNext()) {
+                    fs.delete(path, true);
```

Comment:
This should be done in `SemiTransactionalHiveMetastore`, and only when `commit` is invoked

add a test with:

-`DROP SCHEMA`
- rollback
- try to use that schema